api/v1alpha1/scheduledscaling_types.go

```diff
@@ -0,0 +1,166 @@
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	//Target tortoise phase
+	TortoisePhase TortoisePhase `json:"tortoisePhase" protobuf:"bytes,1,name=tortoisePhase"`
+
+	//unsure if needed
+	ScheduledScalingPhase ScheduledScalingPhase `json:"scheduledScalingPhase" protobuf:"bytes,1,name=scheduledScalingPhase"`
+
+	// AutoscalingPolicy contains the policy how this tortoise actually scales each resource.
+	// It should basically be the same as .spec.autoscalingPolicy.
+	// But, if .spec.autoscalingPolicy is empty, tortoise manages/generates
+	// the policies generated based on HPA and the target deployment.
+	//Commenting this just in case there is a need for separate autoscaling policy in schedule
+	//AutoscalingPolicy []ContainerAutoscalingPolicy `json:"autoscalingPolicy,omitempty" protobuf:"bytes,6,opt,name=autoscalingPolicy"`
+}
+
+// Unsure if needed
+type ScheduledScalingPhase string
+
+type TortoisePhase string
+
+const (
+	// TortoisePhaseInitializing means tortoise is just created and initializing some components (HPA and VPA),
+	// and wait for those components to be ready.
+	TortoisePhaseInitializing TortoisePhase = "Initializing"
+	// TortoisePhaseGatheringData means tortoise is now gathering data and cannot make the accurate recommendations.
+	TortoisePhaseGatheringData TortoisePhase = "GatheringData"
+	// TortoisePhaseWorking means tortoise is making the recommendations,
+	// and applying the recommendation values.
+	TortoisePhaseWorking TortoisePhase = "Working"
+	// TortoisePhasePartlyWorking means tortoise has maxReplicas and minReplicas recommendations ready,
+	// and applying the recommendation values.
+	// But, some of the resources are not scaled due to some reasons. (probably still gathering data)
+	TortoisePhasePartlyWorking TortoisePhase = "PartlyWorking"
+	// TortoisePhaseEmergency means tortoise is in the emergency mode.
+	TortoisePhaseEmergency TortoisePhase = "Emergency"
+	// TortoisePhaseBackToNormal means tortoise was in the emergency mode, and now it's coming back to the normal operation.
+	// During TortoisePhaseBackToNormal, the number of replicas of workloads are gradually reduced to the usual value.
+	TortoisePhaseBackToNormal TortoisePhase = "BackToNormal"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// ScheduledScaling is the Schema for the scheduledscalings API
+type ScheduledScaling struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ScheduledScalingSpec   `json:"spec,omitempty"`
+	Status ScheduledScalingStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// ScheduledScalingList contains a list of ScheduledScaling
+type ScheduledScalingList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ScheduledScaling `json:"items"`
```

Comment:
missing protobuf

Reply:
protobuf is missing in tortoise_types.go as well